x-pack/plugins/triggers_actions_ui/public/application/sections/alert_details/components/alert_details.test.tsx

```diff
@@ -796,6 +796,34 @@ describe('edit button', () => {
   });
 });
 
+describe('refresh button', () => {
+  it('should render alert details when clicked', () => {
+    const alert = mockAlert();
+    const alertType: AlertType = {
+      id: '.noop',
+      name: 'No Op',
+      actionGroups: [{ id: 'default', name: 'Default' }],
+      recoveryActionGroup,
+      actionVariables: { context: [], state: [], params: [] },
+      defaultActionGroupId: 'default',
+      minimumLicenseRequired: 'basic',
+      producer: ALERTS_FEATURE_ID,
+      authorizedConsumers,
+      enabledInLicense: true,
+    };
+
+    expect(
+      shallow(
+        <AlertDetails alert={alert} alertType={alertType} actionTypes={[]} {...mockAlertApis} />
+      )
+        .find(EuiButtonEmpty)
+        .find('[name="refresh"]')
+        .first()
+        .exists()
+    ).toBeTruthy();
```

Comment:
Testing for existence of the button is a great start! Let's also test that the `requestRefresh` button is called when the refresh button is clicked:

```
describe('refresh button', () => {
  it('should call requestRefresh when clicked', () => {
    const alert = mockAlert();
    const alertType: AlertType = {
      id: '.noop',
      name: 'No Op',
      actionGroups: [{ id: 'default', name: 'Default' }],
      recoveryActionGroup,
      actionVariables: { context: [], state: [], params: [] },
      defaultActionGroupId: 'default',
      minimumLicenseRequired: 'basic',
      producer: ALERTS_FEATURE_ID,
      authorizedConsumers,
      enabledInLicense: true,
    };

    const requestRefresh = jest.fn();
    const refreshButton = shallow(
      <AlertDetails
        alert={alert}
        alertType={alertType}
        actionTypes={[]}
        {...mockAlertApis}
        requestRefresh={requestRefresh}
      />
    )
      .find('[data-test-subj="refreshAlertsButton"]')
      .first();

    expect(refreshButton.exists()).toBeTruthy();

    refreshButton.simulate('click');
    expect(requestRefresh).toHaveBeenCalledTimes(1);
  });
});
```